tests/integration-tests/conftest.py

```diff
@@ -1182,18 +1182,19 @@ def odcr_stack(request, region, placement_group_stack, cfn_stacks_factory, vpc_s
     public_subnet = vpc_stack.get_public_subnet()
     public_subnets = vpc_stack.get_all_public_subnets().copy()
     public_subnets.remove(public_subnet)
```

Comment:
We shouldn't do this.

The ODCR stack using the `default_az` is used by single_az tests;
while `public_subnets[0]` and `public_subnets[1]` are used in the MultiAZODCR that is used by multi_az tests.

So it is important that what we here consider `public_subnets[0]` is the same seen by the test while requesting `{{ public_subnets[0] }}`

Reply:
#4898 
Seems like it was a fix applied for a problem encountered making improvements in the test_runner. It appears to ensure that `public_subnets` contains non-default AZs (but the exact reason for doing this is unclear).

Reply:
Likely it could create failures in the code below where we initialize `multi_az_odcr_group` when the two values pointed to the same AZ